lib/services/userpreferences.go

```diff
@@ -31,3 +31,11 @@ type UserPreferences interface {
 	// UpsertUserPreferences creates or updates user preferences for a given username.
 	UpsertUserPreferences(ctx context.Context, req *userpreferencesv1.UpsertUserPreferencesRequest) error
 }
+
+// UserPreferencesBackend is the database interface for managing user preferences.
+type UserPreferencesBackend interface {
```

Comment:
Suggestion: 9ca8e75bb6f269cc8e89d510403d0635e61920cd

Using the same interface for both the gRPC client and server/backend results in this kind of awkward situation situation. It's better to separate them like most of the new endpoints do.

Reply:
Cherry-picked. Thanks